### DISCUSSION TITLE
How can help me to integrate Temp/Hum-Sensor _TZE200_vvmbj46n

### DISCUSSION THREAD

**Discussion by ThomasKlesse** [Koenkk/zigbee2mqtt#18827] [Devices]
I tried to use the Tuya platform to understand the device TZE200_vvmbj46n. But it is not listed among the virtual devices. I only have one SONOFF coordinator, so it's probably difficult for me to work with the Tuya network.
Can you help me to support to integrate the device natively or as a converter file? 

Many thanks!
Thomas

**Comment by Obmawo** [comment]
This works for me. Only the battery state is not working. any help here?

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: tuya.fingerprint('TS0601', ['_TZE200_vvmbj46n']),
    model: 'ZTH01',
    vendor: 'TuYa',
    description: 'Temperature and humidity sensor',
    fromZigbee: [tuya.fz.datapoints, tuya.fz.gateway_connection_status],
    toZigbee: [tuya.tz.datapoints],
    configure: tuya.configureMagicPacket,
    exposes: [e.temperature(), e.humidity(), tuya.exposes.batteryState(), e.battery_low()],
    meta: {
        tuyaDatapoints: [
            [1, 'temperature', tuya.valueConverter.divideBy10],
            [2, 'humidity', tuya.valueConverter.raw],
            [3, 'battery_state', tuya.valueConverter.batteryState],
            // [9, 'temperature_unit', tuya.valueConverter.raw], This DP is not properly supported by the device
        ],
    },
};

module.exports = definition;
```

**Reply by Koenkk** [reply]
Can you enable the z2m debug logging and see if any messages like this pop-up?

```
Datapoint XX not defined for '_TZE200_vvmbj46n'
```

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable debug logging.

**Comment by Obmawo** [comment]
here is some data:

```
Debug 2023-09-07 08:27:52Datapoint 13 not defined for '_TZE200_vvmbj46n' with data {"dp":13,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,20]}}
Debug 2023-09-07 08:27:52Datapoint 17 not defined for '_TZE200_vvmbj46n' with data {"dp":17,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,120]}}
Debug 2023-09-07 08:27:53Datapoint 18 not defined for '_TZE200_vvmbj46n' with data {"dp":18,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,120]}}
Debug 2023-09-07 08:27:53Datapoint 19 not defined for '_TZE200_vvmbj46n' with data {"dp":19,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,6]}}
Debug 2023-09-07 08:27:53Datapoint 20 not defined for '_TZE200_vvmbj46n' with data {"dp":20,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,6]}} 
```

**Comment by ThomasKlesse** [comment]
```
### here data from my log:
debug 2023-09-07 19:42:05: Datapoint 9 not defined for '_TZE200_vvmbj46n' with data {"dp":9,"datatype":4,"data":{"type":"Buffer","data":[0]}}
debug 2023-09-07 19:42:05: Datapoint 10 not defined for '_TZE200_vvmbj46n' with data {"dp":10,"datatype":2,"data":{"type":"Buffer","data":[0,0,1,134]}}
debug 2023-09-07 19:42:06: Datapoint 11 not defined for '_TZE200_vvmbj46n' with data {"dp":11,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,0]}}
debug 2023-09-07 19:42:06: Datapoint 12 not defined for '_TZE200_vvmbj46n' with data {"dp":12,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,60]}}
debug 2023-09-07 19:42:06: Datapoint 13 not defined for '_TZE200_vvmbj46n' with data {"dp":13,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,20]}}
debug 2023-09-07 19:42:06: Datapoint 17 not defined for '_TZE200_vvmbj46n' with data {"dp":17,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,120]}}
debug 2023-09-07 19:42:07: Datapoint 18 not defined for '_TZE200_vvmbj46n' with data {"dp":18,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,120]}}
debug 2023-09-07 19:42:08: Datapoint 19 not defined for '_TZE200_vvmbj46n' with data {"dp":19,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,6]}}
debug 2023-09-07 19:42:08: Datapoint 20 not defined for '_TZE200_vvmbj46n' with data {"dp":20,"datatype":2,"data":{"type":"Buffer","data":[0,0,0,6]}}

```
Thanks Obmawo for your converter

**Comment by Koenkk** [comment]
A lot of extra datapoints, based on the value I cannot immediately determine which on the battery is. Does somebody maybe have a TuYa gateway? It can tell what the values mean: https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html

**Comment by gokkor** [comment]
Try this, partially stolen eeeeeh... inspired from original Tuya.js file:

PS: I apologize but for some reason code block is not working. More than likely its not happy with some chars inside the code but I'm too tired to care right now :)
`
// copied from tuya.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};


const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const legacy = require('zigbee-herdsman-converters/lib/legacy'); // __importStar(require("../lib/legacy"));
const fromZigbee_1 = __importDefault(require("zigbee-herdsman-converters/converters/fromZigbee"));
const toZigbee_1 = require("zigbee-herdsman-converters/converters/toZigbee");

const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: tuya.fingerprint('TS0601', ['_TZE200_vvmbj46n']),
    model: 'TS0601',
    vendor: 'TuYa',
    description: 'Temperature and humidity sensor',
	fromZigbee: [legacy.fromZigbee.nous_lcd_temperature_humidity_sensor, fromZigbee_1.default.ignore_tuya_set_time],
	toZigbee: [legacy.toZigbee.nous_lcd_temperature_humidity_sensor],
	onEvent: tuya.onEventSetLocalTime,
	configure: async (device, coordinatorEndpoint, logger) => {
		const endpoint = device.getEndpoint(1);
		await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
	},
	exposes: [
		e.temperature(), e.humidity(), e.battery(),
		e.numeric('temperature_report_interval', ea.STATE_SET).withUnit('min').withValueMin(5).withValueMax(60).withValueStep(5)
			.withDescription('Temperature Report interval'),
		e.enum('temperature_unit_convert', ea.STATE_SET, ['celsius', 'fahrenheit']).withDescription('Current display unit'),
		e.enum('temperature_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm'])
			.withDescription('Temperature alarm status'),
		e.numeric('max_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60)
			.withDescription('Alarm temperature max'),
		e.numeric('min_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60)
			.withDescription('Alarm temperature min'),
		e.enum('humidity_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm'])
			.withDescription('Humidity alarm status'),
		e.numeric('max_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100)
			.withDescription('Alarm humidity max'),
		e.numeric('min_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100)
			.withDescription('Alarm humidity min'),	
	],
		
};

module.exports = definition;
`

**Reply by gokkor** [reply]
![image](https://github.com/Koenkk/zigbee2mqtt/assets/100941/c51a2417-f8ff-4163-a626-30c782865d2e)

**Comment by ThomasKlesse** [comment]
Thank you very much for you nightwork. I will try it today immediately. Where did you find the "tuya.js"-file? Is it a possible source for another unsupported devices?

**Comment by gokkor** [comment]
It is under the
"/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/lib" folder on my
installation. It may be slightly different for different installations I
presume.

Thank you and best regards,

On Wed, Sep 20, 2023 at 12:26 PM ThomasKlesse ***@***.***>
wrote:

> Thank you very much for you nightwork. I will try it today immediately.
> Where did you find the "tuya.js"-file? Is it a possible source for another
> unsupported devices?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Koenkk/zigbee2mqtt/discussions/18827#discussioncomment-7061333>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAYUTOAMU534O6GCTH7IL3X3MRLVANCNFSM6AAAAAA4H67LXQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Gokhan KORALTURK

**Comment by roastcom** [comment]
> Try this, partially stolen eeeeeh... inspired from original Tuya.js file:

I'm getting a TypeError "e.numeric is not a function" at startup when I 'plug it in', it seems like "e" is not recognised as an object of the correct type - when I comment things out I also get "e.enum is not a function".
Any ideas what's wrong?

```
Using '/app/data' as data directory

/app/data/extension/externally-loaded.js:63

	e.numeric('temperature_report_interval', ea.STATE_SET).withUnit('min').withValueMin(5).withValueMax(60).withValueStep(5).withDescription('Temperature Report interval'),

	  ^

TypeError: e.numeric is not a function

    at /app/data/extension/externally-loaded.js:63:4

    at Script.runInContext (node:vm:141:12)

    at Script.runInNewContext (node:vm:146:17)

    at Object.runInNewContext (node:vm:300:38)

    at loadModuleFromText (/app/lib/util/utils.ts:150:8)

    at loadModuleFromFile (/app/lib/util/utils.ts:157:12)

    at Object.getExternalConvertersDefinitions (/app/lib/util/utils.ts:167:25)

    at getExternalConvertersDefinitions.next (<anonymous>)

    at new ExternalConverters (/app/lib/extension/externalConverters.ts:12:20)

    at new Controller (/app/lib/controller.ts:84:58)
```

**Reply by roastcom** [reply]
Formatted and corrected code that worked for me. Apparently methods needs to be called from "exposes" object, not "e". (In broad terms - I have 0 experience in JS or MQTT :D ).

```
// copied from tuya.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
	if (k2 === undefined) k2 = k;
	var desc = Object.getOwnPropertyDescriptor(m, k);
	if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
		desc = { enumerable: true, get: function() { return m[k]; } };
	}
	Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
	if (k2 === undefined) k2 = k;
	o[k2] = m[k];
}));

var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
	Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
	o["default"] = v;
});

var __importStar = (this && this.__importStar) || function (mod) {
	if (mod && mod.__esModule) return mod;
	var result = {};
	if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
	__setModuleDefault(result, mod);
	return result;
};

var __importDefault = (this && this.__importDefault) || function (mod) {
	return (mod && mod.__esModule) ? mod : { "default": mod };					
};

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const legacy = require('zigbee-herdsman-converters/lib/legacy'); // __importStar(require("../lib/legacy"));
const fromZigbee_1 = __importDefault(require("zigbee-herdsman-converters/converters/fromZigbee"));
const toZigbee_1 = require("zigbee-herdsman-converters/converters/toZigbee");

const e = exposes.presets;
const ea = exposes.access;

const definition = {
	fingerprint: tuya.fingerprint('TS0601', ['_TZE200_vvmbj46n']),
	model: 'TS0601',
	vendor: 'TuYa',
	description: 'Temperature and humidity sensor',
	fromZigbee: [legacy.fromZigbee.nous_lcd_temperature_humidity_sensor, fromZigbee_1.default.ignore_tuya_set_time],
	toZigbee: [legacy.toZigbee.nous_lcd_temperature_humidity_sensor],
	onEvent: tuya.onEventSetLocalTime,
	configure: async (device, coordinatorEndpoint, logger) => {
		const endpoint = device.getEndpoint(1);
		await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
	},
	exposes: [	
		e.temperature(), 
		e.humidity(), 
		e.battery(),
		exposes.numeric('temperature_report_interval', ea.STATE_SET).withUnit('min').withValueMin(5).withValueMax(60).withValueStep(5).withDescription('Temperature Report interval'),
		exposes.enum('temperature_unit_convert', ea.STATE_SET, ['celsius', 'fahrenheit']).withDescription('Current display unit'),
		exposes.enum('temperature_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm']).withDescription('Temperature alarm status'),
		exposes.numeric('max_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60).withDescription('Alarm temperature max'),
		exposes.numeric('min_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60).withDescription('Alarm temperature min'),
		exposes.enum('humidity_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm']).withDescription('Humidity alarm status'),
		exposes.numeric('max_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100).withDescription('Alarm humidity max'),
		exposes.numeric('min_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100).withDescription('Alarm humidity min'),
	],
};

module.exports = definition;
```

**Comment by gokkor** [comment]
That is interesting. "const e = e.presets" line should have seen to that. More than likely it is caused by either a version difference or some of the files are in different paths between our systems and not properly included. In any case I'm glad you've been able to make it work! Have fun !

**Comment by roastcom** [comment]
There isn't "const e = e.presets" in the code though?
I just went over the repository and found a few files with the same methods
and similar structure and copied how they were referenced.

**Reply by gokkor** [reply]
Sorry my friend, my mistake in typing :) it is 'exposes.presets' I did not get my coffee yet :D heheh

`const e = exposes.presets;
const ea = exposes.access;

const definition = {
	fingerprint: tuya.fingerprint('TS0601', ['_TZE200_vvmbj46n']),
`

**Comment by roastcom** [comment]
Well, I'm out of ideas. :man_shrugging: (Not that I had a lot to begin with :laughing: ) Importantly, it works!
Thanks for the code!!

**Reply by gokkor** [reply]
you're welcome!

**Comment by GastonHP** [comment]
I used this code and all was solved for me:

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;
const legacy =require( 'zigbee-herdsman-converters/lib/legacy');

const definition = {
    fingerprint: [{ modelID: 'TS0601', manufacturerName: '_TZE200_vvmbj46n' }],
    model: 'JM-TRH-ZGB-V1',
    vendor: 'TuYa',
    description: 'Temperature & humidity sensor with clock',
    fromZigbee: [legacy.fromZigbee.nous_lcd_temperature_humidity_sensor, fz.ignore_tuya_set_time],
    toZigbee: [legacy.toZigbee.nous_lcd_temperature_humidity_sensor],
    onEvent: tuya.onEventSetLocalTime,
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
    },
    exposes: [
        e.temperature(), e.humidity(), e.battery(),
        e.numeric('temperature_report_interval', ea.STATE_SET).withUnit('min').withValueMin(5).withValueMax(60).withValueStep(5)
            .withDescription('Temperature Report interval'),
        e.enum('temperature_unit_convert', ea.STATE_SET, ['celsius', 'fahrenheit']).withDescription('Current display unit'),
        e.enum('temperature_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm'])
            .withDescription('Temperature alarm status'),
        e.numeric('max_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60)
            .withDescription('Alarm temperature max'),
        e.numeric('min_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60)
            .withDescription('Alarm temperature min'),
        e.enum('humidity_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm'])
            .withDescription('Humidity alarm status'),
        e.numeric('max_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100)
            .withDescription('Alarm humidity max'),
        e.numeric('min_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100)
            .withDescription('Alarm humidity min'),
    ],
};

module.exports = definition;

**Comment by lucassith** [comment]
Do the updates work for you guys? I try to change the slider, I can see in the logs that something is being published, however the state does not change.

**Reply by lucassith** [reply]
I can confirm that state update is possible. My HA setup was a bit croocked and I had to reinstall it.

Nevertheless, I've added humidity report time since it was missing in your scripts.

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;
const legacy = require('zigbee-herdsman-converters/lib/legacy');

const definition = {
  fingerprint: [{
    modelID: 'TS0601',
    manufacturerName: '_TZE200_vvmbj46n'
  }],
  model: 'JM-TRH-ZGB-V1',
  vendor: 'TuYa',
  description: 'Temperature & humidity sensor with clock',
  fromZigbee: [legacy.fromZigbee.nous_lcd_temperature_humidity_sensor, fz.ignore_tuya_set_time],
  toZigbee: [legacy.toZigbee.nous_lcd_temperature_humidity_sensor],
  onEvent: tuya.onEventSetLocalTime,
  configure: async (device, coordinatorEndpoint, logger) => {
    const endpoint = device.getEndpoint(1);
    await reporting.bind(endpoint, coordinatorEndpoint, ['genBasic']);
  },
  exposes: [
    e.temperature(), e.humidity(), e.battery(),
    e.numeric('temperature_report_interval', ea.STATE_SET).withUnit('min').withValueMin(1).withValueMax(60)
    .withValueStep(1)
    .withDescription('Temperature Report interval'),
    e.numeric('humidity_report_interval', ea.STATE_SET).withUnit('min').withValueMin(1).withValueMax(60)
    .withValueStep(1)
    .withDescription('Humidity Report interval'),
    e.enum('temperature_unit_convert', ea.STATE_SET, ['celsius', 'fahrenheit']).withDescription(
      'Current display unit'),
    e.enum('temperature_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm'])
    .withDescription('Temperature alarm status'),
    e.numeric('max_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60)
    .withDescription('Alarm temperature max'),
    e.numeric('min_temperature', ea.STATE_SET).withUnit('°C').withValueMin(-20).withValueMax(60)
    .withDescription('Alarm temperature min'),
    e.enum('humidity_alarm', ea.STATE, ['canceled', 'lower_alarm', 'upper_alarm'])
    .withDescription('Humidity alarm status'),
    e.numeric('max_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100)
    .withDescription('Alarm humidity max'),
    e.numeric('min_humidity', ea.STATE_SET).withUnit('%').withValueMin(0).withValueMax(100)
    .withDescription('Alarm humidity min'),
  ],
};

module.exports = definition;
```

**Comment by gokkor** [comment]
As far as I can remember it works. I'll try and check in the morning.

On Fri, Sep 29, 2023 at 7:07 AM Łukasz Sitarski ***@***.***>
wrote:

> Do the updates work for you guys? I try to change the slider, I can see in
> the logs that something is being published, however the state does not
> change.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Koenkk/zigbee2mqtt/discussions/18827#discussioncomment-7143921>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAYUTOCQTC7OQSW4MFUTULX422ZVANCNFSM6AAAAAA4H67LXQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Gokhan KORALTURK

**Comment by josemrs** [comment]
In my case humidity was showing as Null
I had to add:
`  meta: {
    tuyaDatapoints: [
      [65506, 'humidity', tuya.valueConverter.raw],
    ],
 },`

because I saw this in Z2M debug output

Zigbee2MQTT:debug 2023-11-08 14:28:06: Received Zigbee message from 'THSensor-5-Zigbee-LivingRoom', type 'attributeReport', cluster 'genBasic', data '{"65506":56,"65508":1,"appVersion":72}' from endpoint 1 with groupID 0

**Comment by tolland** [comment]
I've tried a couple of variations, and I see the sensor data in home assistant, but the controls don't work. Error from HA is
```
Nov 25 21:25:47 raspberrypi-00009 hass[55410]: 2023-11-25 21:25:47.293 ERROR (MainThread) 
[homeassistant.components.mqtt.number] Invalid value for number.front_room_temps_temperature_report_interval: 120 (range 1.0 - 60.0)
Nov 25 21:25:47 raspberrypi-00009 hass[55410]: 2023-11-25 21:25:47.294 ERROR (MainThread) 
[homeassistant.components.mqtt.number] Invalid value for number.front_room_temps_humidity_report_interval: 120 (range 1.0 - 60.0)
```

If you change the setting from the zigbee2mqtt device "exposes" form, it does seem to be trying:
```
debug 2023-11-25 21:59:43Received MQTT message on 'zigbee2mqtt/Front_room_Temps/set' with data 
'{"temperature_report_interval":118}'
debug 2023-11-25 21:59:43Publishing 'set' 'temperature_report_interval' to 'Front_room_Temps'
```

However if you try the same thing from HA device page, I get a slightly different error, indicating that HA is using a different topic style to set the value;
```
debug 2023-11-25 22:00:42Received MQTT message on 'zigbee2mqtt/Front_room_Temps/set/humidity_report_interval' with data '98'
debug 2023-11-25 22:00:42Publishing 'set' 'humidity_report_interval' to 'Front_room_Temps'
```

config I am using is this one:
https://gist.github.com/tolland/dc865fce82dd554410a4051273a522ae

I have tried with both `.withValueMax(120)` and `.withValueMax(60)` for those intervals, but no joy.

seems like the state value has been preset to "120" for both humidity_report_interval and temperature_report_interval

**Comment by tolland** [comment]
so the temp and humidity are working fine for me. So pretty happy! ;-)

However I noticed that battery is returning 113%
![2023-11-26_16-34](https://github.com/Koenkk/zigbee2mqtt/assets/91288/e58371bf-19cf-479e-bcde-7dd594d80aa6)

my state from zigbee2mqtt/Front_room_temps is

```json
{
  "battery": 113,
  "humidity": 39,
  "humidity_alarm": "canceled",
  "humidity_report_interval": 120,
  "humidity_sensitivity": 6,
  "linkquality": 127,
  "max_humidity": 60,
  "max_temperature": 39,
  "min_humidity": 20,
  "min_temperature": 0,
  "temperature": 22.3,
  "temperature_alarm": "canceled",
  "temperature_report_interval": 120,
  "temperature_sensitivity": 0.6,
  "temperature_unit_convert": "celsius"
}
```

Also, I am not convinced by the intervals of 120 mins. I see more frequent updates in home assistant, so I would guess that's not being parsed correctly.
